### DISCUSSION TITLE
How to XFAIL for only one particular parameter combination?

### DISCUSSION THREAD

**Discussion by pepoluan** [pytest-dev/pytest#8304] [Q&A]
I'm trying to write a test case like so:

```python
    @pytest.mark.parametrize("init_resp", [True, False])
    @pytest.mark.parametrize("mechanism", ["login", "plain"])
    def test_byclient(self, auth_peeker_controller, client, mechanism, init_resp):
        # self._ehlo contains code that performs EHLO to the server and asserts success
        self._ehlo("example.com")
        client.user = "goodlogin"
        client.password = "goodpasswd"
        auth_meth = getattr(client, "auth_" + mechanism)
        client.auth(mechanism, auth_meth, initial_response_ok=init_resp)
        peeker = auth_peeker_controller.handler
        assert isinstance(peeker, PeekerHandler)
        assert peeker.login == b"goodlogin"
        assert peeker.password == b"goodpasswd"
```

Due to [**bpo-27820**](https://bugs.python.org/issue27820), the combination of `[login-False]` will _always_ fail. Other combinations will work, though.

How do I mark just that one combination as `XFAIL`?

**Comment by bluetech** [comment]
This should answer your question: https://docs.pytest.org/en/stable/example/parametrize.html#set-marks-or-test-id-for-individual-parametrized-test

**Reply by RonnyPfannschmidt** [reply]
unfortunately this implies unrolling the crossproduct

i wish we already had integrated fixture parameter lookup for marker eval

**Reply by pepoluan** [reply]
> This should answer your question: https://docs.pytest.org/en/stable/example/parametrize.html#set-marks-or-test-id-for-individual-parametrized-test

As @RonnyPfannschmidt said above, to do that I have to unroll. Workable for a 2x2 like my situation, but if things get "hairier" in the future, say a 6x4 matrix (of different situation, not str & bool of course), unrolling _just_ to make one particular combination an XFAIL is just ... nasty.

10 values (6+4) vs 24 values...
